src/NServiceBus.Transport.Msmq/DelayedDelivery/DueDelayedMessagePoller.cs

```diff
@@ -0,0 +1,341 @@
+namespace NServiceBus.Transport.Msmq.DelayedDelivery
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Channels;
+    using System.Threading.Tasks;
+    using System.Transactions;
+    using Logging;
+    using Routing;
+    using Unicast.Queuing;
+
+    class DueDelayedMessagePoller
+    {
+        public DueDelayedMessagePoller(
+            MsmqMessageDispatcher dispatcher,
+            IDelayedMessageStore delayedMessageStore,
+            int numberOfRetries,
+            Action<string, Exception, CancellationToken> criticalErrorAction,
+            string timeoutsErrorQueue,
+            Dictionary<string, string> faultMetadata,
+            TransportTransactionMode transportTransactionMode,
+            TimeSpan timeToTriggerFetchCircuitBreaker,
+            TimeSpan timeToTriggerDispatchCircuitBreaker,
+            int maximumRecoveryFailuresPerSecond
+        )
+        {
+            txOption = transportTransactionMode == TransportTransactionMode.TransactionScope
+                ? TransactionScopeOption.Required
+                : TransactionScopeOption.RequiresNew;
+            this.delayedMessageStore = delayedMessageStore;
+            errorQueue = timeoutsErrorQueue;
+            this.faultMetadata = faultMetadata;
+            this.numberOfRetries = numberOfRetries;
+            this.dispatcher = dispatcher;
+            fetchCircuitBreaker = new RepeatedFailuresOverTimeCircuitBreaker("MsmqDelayedMessageFetch", timeToTriggerFetchCircuitBreaker,
+                ex => criticalErrorAction("Failed to fetch due delayed messages from the storage", ex, CancellationToken.None));
```

Comment:
Mmm, it feels CancellationToken.None is not the right value. Shouldn't this be forwarded based on hosting settings?

Reply:
SQLT passes there a token that is used to stop its message pump so I guess we should pass the poller's `tokenStouce.Token` here.